iroh-p2p/src/rpc.rs

```diff
@@ -394,6 +395,33 @@ impl P2p {
         Ok(peer_info_from_lookup(lookup))
     }
 
+    #[tracing::instrument(skip(self))]
+    fn network_events(
+        self,
+        _: NetworkEventsRequest,
+    ) -> BoxStream<'static, Box<NetworkEventsResponse>> {
+        async move { self.network_events_0().await }
+            .flatten_stream()
+            .boxed()
+    }
+
+    #[tracing::instrument(skip(self))]
+    async fn network_events_0(self) -> impl Stream<Item = Box<NetworkEventsResponse>> {
+        let (s, r) = oneshot::channel();
+        self.sender
+            .send(RpcMessage::NetworkEvents(s))
+            .await
+            // TODO(ramfox): refactor to return result
+            .expect("expect sender to exist");
+
+        let mut r = r.await.expect("expect sender and receiver to still exist");
```

Comment:
Here too `.expect()` is probably too harsh.  If the remote server closes the connection before there's a response, is the sender not dropped?